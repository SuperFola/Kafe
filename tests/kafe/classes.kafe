x: int = bar.foo()
x: int = bar.foo(1)
x: int = bar.foo(1, -4)
x: int = bar.foo(test())
x: int = bar.foo(bar.egg())

cls Test
    new Test()
        m_x = 10
    end

    fun hiThere(x: int) -> int
        ret (x * 2)
    end
end

lol: Test = new Test()
