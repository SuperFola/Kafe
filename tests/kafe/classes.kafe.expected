(Program
    (Definition
        (VarName x)
        (Type int)
        (MethodCall
            (ClassName bar)
            (FuncName foo)
            (Args)
        )
    )
    (Definition
        (VarName x)
        (Type int)
        (MethodCall
            (ClassName bar)
            (FuncName foo)
            (Args
                (Integer 1)
            )
        )
    )
    (Definition
        (VarName x)
        (Type int)
        (MethodCall
            (ClassName bar)
            (FuncName foo)
            (Args
                (Integer 1)
                (Integer -4)
            )
        )
    )
    (Definition
        (VarName x)
        (Type int)
        (MethodCall
            (ClassName bar)
            (FuncName foo)
            (Args
                (FunctionCall
                    (Name test)
                    (Args)
                )
            )
        )
    )
    (Definition
        (VarName x)
        (Type int)
        (MethodCall
            (ClassName bar)
            (FuncName foo)
            (Args
                (MethodCall
                    (ClassName bar)
                    (FuncName egg)
                    (Args)
                )
            )
        )
    )
    (Class
        (Name Test)
        (ClassConstructor
            (Name Test)
            (Arguments)
            (Body
                (Assignment
                    (Name m_x)
                    (Integer 10)
                )
            )
        )
        (Body
            (Function
                (Name hiThere)
                (Arguments
                    (Declaration
                        (Name x)
                        (Type int)
                    )
                )
                (Type int)
                (Body
                    (Ret
                        (OperationsList
                            (VarUse x)
                            (Operator *)
                            (Integer 2)
                        )
                    )
                )
            )
        )
    )
    (Definition
        (VarName lol)
        (Type Test)
        (ClassInstanciation
            (Name Test)
            (Args)
        )
    )
)